extensions/sql-migration/src/api/azure.ts

```diff
@@ -640,8 +816,100 @@ export interface StartDatabaseMigrationRequest {
 
 export interface StartDatabaseMigrationResponse {
 	status: number,
-	databaseMigration: DatabaseMigration
-	asyncUrl: string
+	databaseMigration: DatabaseMigration,
+	asyncUrl: string,
+}
+
+export enum Kind {
+	SQLDB = 'SqlDb',
+	SQLMI = 'SqlMi',
+	SQLVM = 'SqlVm',
+}
+
+export interface validateIrSqlDatabaseMigraionRequest {
+	sourceDatabaseName: string,
+	targetDatabaseName: string,
+	kind: string,
+	validateIntegrationRuntimeOnline?: boolean,
+	sourceSqlConnection: {
+		testConnectivity?: boolean,
+		dataSource: string,
+		userName: string,
+		password: string,
+		encryptConnection?: boolean,
+		trustServerCertificate?: boolean,
+		authentication: string,
+	},
+	targetSqlConnection: {
+		testConnectivity?: boolean,
+		dataSource: string,
+		userName: string,
+		password: string,
+		encryptConnection?: boolean,
+		trustServerCertificate?: boolean,
+		authentication: string,
+	},
+}
+
+export interface ValdiateIrDatabaseMigrationRequest {
+	sourceDatabaseName: string,
+	kind: string,
+	validateIntegrationRuntimeOnline?: boolean,
+	backupConfiguration: {
+		targetLocation: {
+			testConnectivity?: boolean,
+			storageAccountResourceId?: string,
+			accountKey?: string,
+		},
+		sourceLocation: {
+			testConnectivity?: boolean,
+			fileShare: {
+				path: string,
+				username: string,
+				password: string,
+			}
+		}
+	},
+	sourceSqlConnection: {
+		testConnectivity?: boolean,
+		dataSource?: string,
+		userName?: string,
+		password?: string,
+		encryptConnection?: boolean,
+		trustServerCertificate?: boolean,
+		authentication?: string,
+	},
+}
+
+export interface ValidationError {
+	code: string,
+	message: string,
+}
+
+export interface ValdiateIrDatabaseMigrationResponse {
+	kind: string,
+	sourceDatabaseName: string,
+	sourceSqlConnection: {
+		testConnectivity: boolean,
+		"encryptConnection": true,
```

Comment:
why are `encryptConnection` and `trustServerCertificate` in quotes but the others aren't?